electron/src/trusted-main-electron-process/SeedableHardwareKeys/IpcServer.ts

```diff
@@ -0,0 +1,100 @@
+import {Server, Socket} from "net";
+import * as net from "net"
+import * as path from "path"
+import * as sudo from "sudo-prompt"
+import * as child_process from 'child_process'
+import fs from "fs";
+import {
+    DeviceListUpdateCallback, ErrorCallback,
+    StopMonitoringFunction
+} from "./UsbDeviceMonitor";
+import {ExecException} from "child_process";
+import {DeviceUniqueIdentifier} from "../../../../common/IElectronBridge";
+import {isWin} from "../../usb";
+
+export var UsbSocketPath : string
+
+if(process.platform == "win32"){
+    UsbSocketPath = path.join('\\\\?\\pipe', process.cwd(), 'dicekeys')
+} else {
+    UsbSocketPath = 'usb.sock'
+}
+
+function escapeParamCmd(value: any): string {
+    // Make sure it's a string
+    // Escape " -> \"
+    // Surround with double quotes
+    return `"${String(value).replace(/"/g, '\\"')}"`;
+}
+let ipcServer: Server | null
+let clientSocket: Socket | null
+export function createIpcServer(deviceListUpdateCallback: DeviceListUpdateCallback, errorCallback?: ErrorCallback) : StopMonitoringFunction{
+    ipcServer = net.createServer((socket: Socket) => {
+        clientSocket = socket
+
+        socket.on('data', (data => {
+            let json = JSON.parse(data.toString())
+
+            if (json.error) {
+                errorCallback?.(json.error)
+            } else if (json.devices) {
+                deviceListUpdateCallback(json.devices)
+            }
+        }))
+    });
+
+    if(!isWin){
+        // clear socket file
+        if(fs.existsSync(UsbSocketPath))  fs.unlinkSync(UsbSocketPath)
+    }
+
+    console.log("Starting IPC Server", UsbSocketPath)
+
+    ipcServer.listen(UsbSocketPath, () => {
+        let cmd = escapeParamCmd(process.argv[0]) + ' ' + path.join('dist', 'src', 'usb-writer.js')
+
+        const callback = (error?: Error | ExecException | null, stdout?: string | Buffer, stderr?: string | Buffer) => {
+            console.log('usb-writer terminated')
+            console.log(error)
+            console.log(stdout)
+            console.log(stderr)
+        }
+
+        const env = {
+            ELECTRON_RUN_AS_NODE: '1',
+        }
+
+        // Elevate
+        if(isWin){
+            console.log("Elavate", cmd)
+            sudo.exec(cmd, {
+                name: 'DiceKeys',
+                env,
+            }, callback)
+        }else{
+            // on macos / linux run it a a regular exec
+            console.log("Spawn", cmd)
```

Comment:
Why spawn instead of running in-process on those platforms?

Reply:
Used only while developing on macos to have the same behavior as windows.

Reply:
Ah.  How about a `runUsbCommandsInSeparateProcess` flag, set to windows by default, and with a note that it can be used for debugging on other platforms?

Reply:
Fixed